modules/@apostrophecms/attachment/index.js

```diff
@@ -87,6 +87,28 @@ module.exports = {
       }
     ];
 
+    if (self.options.addFileGroups) {
+      self.options.addFileGroups.forEach(newGroup => {
+        if (self.fileGroups.some(existingGroup => existingGroup.name === newGroup.name)) {
+          const existingGroup = self.fileGroups.find(existingGroup => existingGroup.name === newGroup.name);
+          if (newGroup.extensions) {
+            existingGroup.extensions = [ ...existingGroup.extensions, ...newGroup.extensions ];
+          };
+          if (newGroup.extensionMaps) {
+            existingGroup.extensionMaps = {
+              ...existingGroup.extensionMaps,
+              ...newGroup.extensionMaps
+            };
+          }
+          const groupIndex = self.fileGroups.findIndex(existingGroup => existingGroup.name === newGroup.name);
+          self.fileGroups.splice(groupIndex, 1);
+          self.fileGroups.push(existingGroup);
```

Comment:
The order of the groups doesn't matter, they are never enumerated in the UI.

Reply:
If you factor this inner bit out to an `addFileGroup` method, then we'll have something that modules can call to programmatically add a file group during initialization without breaking project level use of the `addFileGroups` option.

Reply:
Re: the order. I did know that the order didn't matter. Embarrassingly, I did not know that find mutated in place. I thought it returned a new object that had to be used to replace the original, hence the splice and push.

Re: refactoring. I factored it out (I hope I understood the ask), but I wouldn't mind an explanation of "without breaking project level use of the addFileGroups option". Not sure why factoring out was needed.